### PR TITLE
End to End testing: student side

### PR DESCRIPTION
**Pull request by A-M-00**
# Lesson lifecycle unreliable test fixes and code clarity improvements

## Overview
This week's PR, meant to implement issue #190, includes two end to end tests meant to test the student side of the app across various screens of account and lesson creation

### 1. **endToEndStudentCreateAccountAndLesson** 
- This test creates a new student account, then creates, edits, and deletes a lesson request. It then tests displays for some other lesson types.

### 2.  **endToEndStudentEditAccountAndLessonLifecycle** 
- This test creates a new student account, then edits it, then creates a new instant lesson.

### 3. **Clean up Warnings**
- Various code warnings given by the IDE in various screens touched by the tests have been cleaned up to improve code quality, and make it correspond better to accepted coding standards

### 3. **Refactor strings**
- Following TA feedback, some strings in ProfileInfo have been refactored into strings.xml

---

## How to Test

1. Run the CI
2. Verify that the corrected tests pass
3. Verify the strings are no longer hardcoded
4. Verify that the compiler and IDE give no warnings for those files

### CODE REVIEW

**Review by MPepito — COMMENTED** [review]
Please merge the main and solve conflicts as it seems you have code in green in your PR that comes from Philipp's PR that got merged now. Also I doubt that `endToEndStudentCreateAccountEditAccountAndInstantLesson` would succed the CI as there are some Thread.sleep and many wait and as we have seen multiple times it most of the time make the CI to fail. Also please adjust the few lines where `.isNotDisplayed()`is used instead of `.isDisplayed()`

File: app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt [hunk 1]
```diff
@@ -360,5 +317,100 @@ class EndToEndTest {
...
+      composeTestRule.onNodeWithTag("instantButton").isDisplayed()
+    }
+
+    // Set Title and Description
+    composeTestRule.onNodeWithTag("titleField").performTextInput("Math Lesson")
+    composeTestRule.onNodeWithTag("DescriptionField").performTextInput("This is a math lesson.")
+
+    composeTestRule.waitForIdle()
+    composeTestRule.onNodeWithTag("instantButton").performClick()
+    composeTestRule.waitUntil(20000) { composeTestRule.onNodeWithTag("mapButton").isNotDisplayed() }
```
[the remark this answers is not in the document]
Comment (A-M-00) [reply app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt:380]: No, if the lesson is instant, the button to select your location should disappear. This is a way to check that the lesson correctly went instant

File: app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt [hunk 2]
```diff
@@ -221,7 +221,6 @@ class EndToEndStudentTest {
       testMapReady
     }
     composeTestRule.waitUntil(20000) { composeTestRule.onNodeWithTag("googleMap").isDisplayed() }
-    Thread.sleep(100)
```
Comment (MPepito) [inline app/src/androidTest/java/com/github/se/project/ui/endToEnd/StudentEndToEnd.kt]: I guess that might help the CI to fail less often, well done

**Review by MPepito — APPROVED** [review]
Good adjustment, I won't repeat the comments I wrote in my first review just above so well done!

File: app/src/androidTest/java/com/github/se/project/ui/lesson/AddLessonTest.kt [hunk 3]
```diff
@@ -291,7 +291,6 @@ class AddLessonTest {
     }
 
     composeTestRule.onNodeWithTag("confirmButton").performClick()
-    verify(navigationActions).navigateTo(anyString())
```
Comment (MPepito) [inline app/src/androidTest/java/com/github/se/project/ui/lesson/AddLessonTest.kt:294]: Thanks for removing that line that constantly fail the CI

**Review by MPepito — APPROVED** [review]
Thanks for these adjustment following my first review

**Pull request merged**